lib/jekyll/tags/highlight.rb

```diff
@@ -14,22 +14,9 @@ def initialize(tag_name, markup, tokens)
         super
         if markup.strip =~ SYNTAX
           @lang = Regexp.last_match(1).downcase
-          @highlight_options = {}
-          if defined?(Regexp.last_match(2)) && Regexp.last_match(2) != ''
-            # Split along 3 possible forms -- key="<quoted list>", key=value, or key
-            Regexp.last_match(2).scan(/(?:\w="[^"]*"|\w=\w|\w)+/) do |opt|
-              key, value = opt.split('=')
-              # If a quoted list, convert to array
-              if value && value.include?("\"")
-                value.delete!('"')
-                  value = value.split
-              end
-              @highlight_options[key.to_sym] = value || true
-            end
-          end
-          @highlight_options[:linenos] = "inline" if @highlight_options.key?(:linenos) && @highlight_options[:linenos] == true
+          @highlight_options = parse_options(Regexp.last_match(2))
         else
-          raise SyntaxError.new <<-eos
+          raise SyntaxError, <<-eos
```

Comment:
I think we provide a heredoc stripper in the Utils class so that you can indent.  I digress though on it's usage, I like to use it but if you prefer not to I'm okay with that too!